source/transactions-convenient-api/tests/commit-writeconcernerror.yml

```diff
@@ -193,3 +193,24 @@ tests:
     expectations: *expectations_without_retries
     # failCommand with writeConcernError still applies the write operation(s)
     outcome: *outcome
+
+  -
+    description: commitTransaction is not retried after MaxTimeMSExpired error
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 1 }
+      data:
+          failCommands: ["commitTransaction"]
+          writeConcernError:
+            code: 50
+            codeName: MaxTimeMSExpired
+            errmsg: "operation exceeded time limit"
+    operations:
+      - <<: *operation
+        result:
+          errorCodeName: MaxTimeMSExpired
+          errorLabelsContain: ["UnknownTransactionCommitResult"]
+          errorLabelsOmit: ["TransientTransactionError"]
+    expectations: *expectations_without_retries
+    # failCommand with writeConcernError still applies the write operation(s)
```

Comment:
Under what circumstances would MaxTimeMSExpired get reported via a write concern error? I'm curious how that differs from a write concern timeout.

Is trusting that the write is still applied just something unique to using the fail point? I assume in a real world scenario, we wouldn't know whether (or how much of) the write might have been applied.

Reply:
> Under what circumstances would MaxTimeMSExpired get reported via a write concern error? I'm curious how that differs from a write concern timeout.

wtimeout only applies when the server is waiting for a writeConcern. maxTimeMS applies to the entire command execution (including the writeConcern wait). So when maxTimeMS is exceeded while waiting for writeConcern the server will return it as a writeConcernError:  ` {ok:1, writeConcernError: {code: 50, codeName: "MaxTimeMSExpired"}}`.

> Is trusting that the write is still applied just something unique to using the fail point? I assume in a real world scenario, we wouldn't know whether (or how much of) the write might have been applied.

It is not unique to the failpoint and this would also be the case in a real world scenario. MaxTimeMSExpired in a writeConcernError means that the write was applied on the primary but the given writeConcern could not be satisfied within the remaining maxTimeMS. We still label this  UnknownTransactionCommitResult because we don't know if the commit has satisfied the write concern; we may later find out that the primary stepped down and the write was rolled back or we may find out that the write succeeded on a majority.

Reply:
Thanks for the helpful explanation!